src/mtcp/mtcp_restart.c

```diff
@@ -864,6 +865,10 @@ static int read_one_memory_area(int fd)
         imagefd = mtcp_sys_open (area.name, O_CREAT|O_RDWR, 0);
         if (imagefd >= 0) {
           mtcp_sys_unlink(area.name);
+        } else if (fix_parent_directories(area.name) == area.name) {
```

Comment:
I don't want to put it ahead of mtcp_sys_open().  Primarily, I was thinking about efficiency.  fix_parent_directories is not necessarily cheap, since it starts at the root directory and tries to add each subdirectory in turn, testing for EEXIST.  Hence, I call it only if mtcp_sys_open() fails.

However, the point about creating dead code is excellent.  Nice catch!  I'll change the logic so that after calling fix_parent_directories(), I'll again try to open the file, and if I still fail, then I'll follow the existing "else" condition and change from MAP_SHARED to MAP_PRIVATE.